central/vulnmgmt/vulnerabilityrequest/manager/requestmgr/manager_impl.go

```diff
@@ -448,17 +489,31 @@ func (m *managerImpl) reprocessDeployments(requestID string, affectedImages ...s
 func (m *managerImpl) reprocessImage(_ string, affectedImages ...string) error {
 	imageKeys := make([]*central.InvalidateImageCache_ImageKey, 0, len(affectedImages))
 	for _, imgID := range affectedImages {
-		image, found, err := m.images.GetImage(allAccessCtx, imgID)
-		if err != nil {
-			return errors.Wrap(err, "could not get image for reprocessing")
-		}
-		if !found {
-			continue
+		if features.FlattenImageData.Enabled() {
+			image, found, err := m.imagesV2.GetImage(allAccessCtx, imgID)
+			if err != nil {
+				return errors.Wrap(err, "could not get image for reprocessing")
+			}
+			if !found {
+				continue
+			}
+			imageKeys = append(imageKeys, &central.InvalidateImageCache_ImageKey{
```

Comment:
This is a tricky one. These keys would be new imageIDs, but if sensor is still in older version, it wouldn't recognize the new format keys. You can leave this for now and create a followup ticket to revisit this stuff.